ivy/functional/backends/torch/experimental/manipulation.py

```diff
@@ -358,3 +359,105 @@ def unique_consecutive(
         inverse_indices,
         counts,
     )
+  
+  
+def _can_handle_padding(input_shape, pad, mode):
+    n_dims = len(input_shape)
+    if n_dims == 0:
+        return False
+    if n_dims == 1 and mode != 'circular':
+        return False
+    if mode == 'circular':
+        return True
+    if isinstance(pad, int):
+        return False
+    for i in range(n_dims):
+        if not isinstance(pad[2 * i], int) or not isinstance(pad[2 * i + 1], int):
+            return False
+        if input_shape[i] + pad[2*i] + pad[2*i+1] < 0:
+            return False
+    return True
```

Comment:
To be more specific on the point I was trying to make about converting the padding to the correct format so we can call the native functions in more cases:
Here, if the padding is not a tuple of integers, the function returns False. However, if the padding is an integer or a tuple of tuples (as expected by ivy.pad) you can just convert it to a tuple of integers like the one torch is expecting (as shown in the example code I wrote). 
Have you tried debugging to see if the code ever gets inside the torch backend during ivy pad's test?

Reply:
1) I have made some changes that needed to be in the `_can_handle_padding` helper function of the pad function in the torch backend, I also added an extra check at the beginning of the `_can_handle_padding` helper function which checks `whether it is tuple of tuple then convert it into a tuple of integers`. 

2) I did not add the integer case that you suggested, the reason is that when I tested it then it didn't return the same value when we convert the integer into a tuple and apply the padding function.

3) I have one question about the implementation of the torch pad function: while I was testing some examples in `torch.nn.functional.pad` which is fully native and I find out that the output of the torch backend is different then the output of the numpy backend. 

Here is this example: 
Torch backend:
`inp = torch.tensor([[1., 2., 3.], [4., 5., 6.], [7., 8., 9.]])
pad = (1,1)
mode='constant' 
res = torch.nn.functional.pad(inp, pad, mode, value=0)`

res: 
`tensor([[0., 1., 2., 3., 0.],
        [0., 4., 5., 6., 0.],
        [0., 7., 8., 9., 0.]])`

Numpy Backend:

`array = np.array([[1., 2., 3.], [4., 5., 6.], [7., 8., 9.]])
pad_width = (1,1)
mode='constant'

res = np.pad(array, pad_width, mode)`

res: 

`array([[0., 0., 0., 0., 0.],
       [0., 1., 2., 3., 0.],
       [0., 4., 5., 6., 0.],
       [0., 7., 8., 9., 0.],
       [0., 0., 0., 0., 0.]])`

so, I find out in the docs of torch: [Docs](https://pytorch.org/docs/stable/generated/torch.nn.functional.pad.html) about torch applying padding in a bit different way, it has `pad = (padding_left, padding_right, padding_top, padding_bottom, padding_front, padding_back)padding_front,padding_back)`. additionally when I converted pad=(1,1) to pad=(1,1,1,1) in torch backend for testing purpose then it returns the same answer as numpy backend returns when the input's pad=(1,1) in numpy. 

Now I'm stuck on this point if we directly call the native implementation ( when it is needed to call the native one ) in the torch backend but it returns the answer which does not match the desirable output because of the internal functionality of the torch's pad function. then this becomes irrelevant. So, what can we do now? any suggestions? 

I think we should keep the compositional implementation for these test cases, it may be inefficient but it can provide users the answers they want, but what do you think about it @AnnaTz?

Reply:
It is fairly simple to convert the pad widths to torch's expected format inside the torch backend implementation. Please have a look at how our torch frontend does the inverse of this (converting pad widths from torch format to ivy/numpy format). A similar approach should be followed to handle the integer pad width cases.